### PR TITLE
added debounce function for search, installed lodash-es for debounce

### PR DESCRIPTION
**Pull request by fikfan**
# added debounce function to improve search load times/overall _smoothness_

**tl;dr**
- installed `lodash-es` package
- added a debounce method in `handleSearch` in `page.tsx` to improve search load times

**issue**
- experiencing slight lag when searching masjid/institution names
- same thing when clearing the search bar

**possible causes**
- `handleSearch` in `page.tsx` being called on every input, lead to frequent re-render/re-filter(mungkin)

**solution**
- added a debounce function to improve _smoothness_ [this article explains debounce way better than i do ](https://dev.to/jeetvora331/javascript-debounce-easiest-explanation--29hc)
- basically, implemented a function to add a slight delay (300ms) with each input, so it won't call the search function every time a user types a character - refer to `debouncedSetQuery` in `page.tsx`

**outcomes**
- smoother user experience with reduced lag when searching & clearing input
- can facilitate bigger data size later on(hopefully it doesn't lag when it reaches 10k LOC in `institution.tsx`)

**additional improvisation comments**
- maybe load only up to 40 masjid in 1 page? because `page.tsx` renders all data in 1 instantaneous load, so it might contribute to the lag i guess

feel free to review and let me know if you have any questions or improvements! 

loving the simple idea of this project so far!
my first pr(publicly) dont kill me plz

### CODE REVIEW

**Comment by khrnchn** [comment]
thanks for the pr! and also the debounce explanation. i have tested it out and yes it feels smoother than before. saya kentang je, tak reti nak review-review ni haha. as long as it lgtm then ill straightaway merge xd

> maybe load only up to 40 masjid in 1 page? because page.tsx renders all data in 1 instantaneous load, so it might contribute to the lag i guess

also someone else brought this up too, so i think its a good suggestion!
